corehq/apps/app_manager/static/app_manager/js/forms/case_config_ui.js

```diff
@@ -317,6 +317,28 @@ hqDefine('app_manager/js/forms/case_config_ui', function () {
                 self.case_name = null;
             }
 
+            // Pagination and search
+            self.case_property_query = ko.observable('');
+            self.filtered_case_properties = ko.computed(function () {
+                return _.filter(self.case_properties(), function (item) {
+                    return item.path().indexOf(self.case_property_query()) !== -1;
+                });
+            });
+            self.visible_case_properties = ko.observableArray();
+            self.goToPage = function (page) {
+                page = page || 1;
+                var props = self.filtered_case_properties();
+                var skip = self.per_page() * (page - 1);
+                props = props.slice(skip, skip + self.per_page());
+                self.visible_case_properties(props);
+            };
+            // Don't allow changing per page; "Add Property" button is where the per page changer usually is
+            self.per_page = ko.observable(25);
```

Comment:
nit: should these be CamelCase?

Reply:
eh...I don't know. The casing in this file is a mixed bag since everything coming from python is snake case and then properties created in js are mostly but not entirely camel case. I'm willing to change it if you think it'd help readability, but if you don't care much I'm going to leave it alone.